server/svix-server/src/v1/endpoints/health.rs

```diff
@@ -1,12 +1,88 @@
 // SPDX-FileCopyrightText: © 2022 Svix Authors
 // SPDX-License-Identifier: MIT
 
-use axum::{http::StatusCode, routing::get, Router};
+use std::time::Duration;
 
-async fn health() -> StatusCode {
+use axum::{http::StatusCode, routing::get, Extension, Json, Router};
+use sea_orm::{query::Statement, ConnectionTrait, DatabaseBackend, DatabaseConnection};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    core::cache::{kv_def, Cache, CacheBehavior, CacheKey, CacheValue},
+    queue::{QueueTask, TaskQueueProducer},
+};
+
+async fn heartbeat() -> StatusCode {
     StatusCode::NO_CONTENT
 }
 
+#[derive(Debug, Deserialize, Serialize)]
+pub enum HealthStatus {
+    Operational,
+    Error(String),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct HealthReport {
+    pg_status: HealthStatus,
+
+    queue_status: HealthStatus,
+    cache_status: HealthStatus,
+}
+
+#[derive(Deserialize, Serialize, Debug, PartialEq)]
+struct HealthCheckCacheValue(());
+kv_def!(HealthCheckCacheKey, HealthCheckCacheValue);
+
+async fn health(
+    Extension(pg): Extension<DatabaseConnection>,
+    Extension(queue_tx): Extension<TaskQueueProducer>,
+    Extension(cache): Extension<Cache>,
+) -> Json<HealthReport> {
+    // SELECT 1 FROM any table
+    let pg_status = if let Err(e) = pg
+        .execute(Statement::from_string(
+            DatabaseBackend::Postgres,
+            "SELECT 1 FROM endpoint".to_owned(),
```

Comment:
```suggestion
            "SELECT 1".to_owned(),
```
This is sufficient. :)